server/channels/audit/record.go

```diff
@@ -121,3 +121,10 @@ func (rec *Record) AddErrorCode(code int) {
 func (rec *Record) AddErrorDesc(description string) {
 	rec.Error.Description = description
 }
+
+// AddToEventActor adds the userId and sessionId to the audit record's Actor. Intended for when the
+// initial call to context.MakeAuditRecord did not have the userId and sessionId available.
+func (rec *Record) AddToEventActor(userId, sessionId string) {
```

Comment:
The userId and sessionId are the only two that might not be available when `MakeAuditRecord` is called, so they are the only two that would need to be updated.

Reply:
The only alternative I could think of here would be to split these two into `AddUserId` and `AddSessionId`, so that the order of the parameters is never mixed (I'm always hypervigilant when I have to pass two or more adjacent arguments of the same type to a function)

Reply:
Yep, that makes sense, and it's clearer.

Reply:
A couple of questions:

- Have we considered using `web.(*Context).LogAuditRec()` to backfill what's missing from the available context? This could be useful for more than this specific endpoint.
- Why do we need methods to set when all the structs involved are public?

Reply:
@streamer45 
1. It took me a bit to make sure that we wouldn't be adding something that we didn't want to, but it looks good, thanks for the idea.
2. Although it's moot now, the reason was that 1) we're already setting all these public fields with methods, so I wanted to be consistent, 2) it makes sense to set them with methods to keep all that logic together, rather than having them set silently all over the place. (Then why are they public? I imagine it's for serialization. One point off for Go.)

Reply:
> It took me a bit to make sure that we wouldn't be adding something that we didn't want to

Thanks for taking the time. These are audit logs, so I can't imagine an issue with adding user/session IDs to them (when missing that is). They are meant to be there whenever available, I suppose.


/cc @agarciamontoro @enzowritescode in case you have concerns with the new approach.

Reply:
I'll rerequest @enzowritescode considering it's a significant change.